src/datajudge/utils.py

```diff
@@ -42,3 +42,138 @@ def format_difference(
         f"{s1[:diff_idx]}{_fmt_diff_part(s1, diff_idx)}",
         f"{s2[:diff_idx]}{_fmt_diff_part(s2, diff_idx)}",
     )
+
+
+class OutputProcessor(Protocol):
+    def __call__(  # noqa: E704
+        self,
+        collection: Collection,
+        counts: Optional[Collection] = None,
+    ) -> Collection: ...
+
+
+def output_processor_sort(
+    collection: Collection, counts: Optional[Collection] = None
+) -> Collection:
+    """
+    Sorts a collection of tuple elements in descending order of their counts,
+    and for ties, makes use of the ascending order of the elements themselves.
+
+    If the first element is not instanceof tuple,
+    each element will be transparently packaged into a 1-tuple for processing;
+    this process is not visible to the caller.
+
+    Handles None values as described in `sort_tuple_none_aware`.
+    """
+    collection = list(collection)
+    if not isinstance(collection[0], tuple):
+        # package into a 1 tuple and pass into the method again
+        packaged_list = [(elem,) for elem in collection]
+        res_main, res_counts = output_processor_sort(packaged_list, counts)
+        return [elem[0] for elem in res_main], res_counts
+
+    if counts is None:
+        return sort_tuple_none_aware(collection), counts
+
+    assert len(collection) == len(
```

Comment:
I would advocate against using assert for anything that is neither
* debugging nor
* testing.
Instead, I would suggest to explicitly raise Exceptions, in this case for instance a `ValueError`. 

The reason behind this is that `assert` is not actually guaranteed to be executed; it can be compiled away:
https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement